test/location.test.js

```diff
@@ -13,6 +18,32 @@ define(function(require) {
       window.history.replaceState(null, null, originalPath)
     })
 
+    describe('Instantiation', function() {
+      describe('upgrading from hash path', function() {
+        it('should preserve query parameters', function() {
+          var loc = window.location
+          var originalQuery = new URLSearchParams()
+
+          originalQuery.set('foo', 'bar')
+          originalQuery.set('fizz', 'buzz')
+
+          window.history.replaceState(
+            null, null,
+            loc.origin + loc.pathname +
+            '?' + originalQuery.toString() +
+            '#/app/workspaces'
```

Comment:
I'd prefer it if we could use explicit constants wherever possible to make these tests more readable. So instead of all this string building, just `/?foo=bar&fizz=buzz#/app/workspaces` or whatever is relevant test data.